### PR TITLE
makes fuse an optional dependency and fix wedit

### PR DESCRIPTION
**Pull request by maxwang7**
@fabeschan @percyliang

### CODE REVIEW

**Comment by maxwang7** [comment]
Tested on an Ubuntu instance with fuse uninstalled.

**Comment by kashizui** [comment]
Does this also work for the case where fuse.py is installed, but libfuse isn't?

File: codalab/lib/bundle_cli.py [hunk 1]
```diff
@@ -1756,7 +1756,8 @@ def do_mount_command(self, args):
         path_util.check_isvalid(mountpoint, 'mount')
         print >>self.stdout, 'BundleFUSE mounting bundle {} on {}'.format(uuid, mountpoint)
         print >>self.stdout, 'BundleFUSE will run and maintain the mounted filesystem in the foreground. CTRL-C to cancel.'
-        bundle_fuse.bundle_mount(client, mountpoint, target, args.verbose)
+        if not bundle_fuse.fuse_is_available:
```
Comment (fabeschan) [inline codalab/lib/bundle_cli.py]: shouldn't this be `if bundle_fuse.fuse_is_available`?

**Comment by fabeschan** [comment]
Print an instruction when fuse is not available and someone is attempting to call mount

File: codalab/lib/bundle_cli.py [hunk 2]
```diff
@@ -1756,7 +1756,8 @@ def do_mount_command(self, args):
         path_util.check_isvalid(mountpoint, 'mount')
         print >>self.stdout, 'BundleFUSE mounting bundle {} on {}'.format(uuid, mountpoint)
         print >>self.stdout, 'BundleFUSE will run and maintain the mounted filesystem in the foreground. CTRL-C to cancel.'
```
Comment (fabeschan) [inline codalab/lib/bundle_cli.py]: perhaps put these print statements inside the if-clause

File: codalab/lib/bundle_cli.py [hunk 3]
```diff
@@ -1756,7 +1756,8 @@ def do_mount_command(self, args):
...
-        bundle_fuse.bundle_mount(client, mountpoint, target, args.verbose)
+        if not bundle_fuse.fuse_is_available:
+            bundle_fuse.bundle_mount(client, mountpoint, target, args.verbose)
         print >>self.stdout, 'BundleFUSE shutting down.'
```
Comment (fabeschan) [inline codalab/lib/bundle_cli.py]: ditto

**Comment by maxwang7** [comment]
Made changes, thanks for the comments!

Needs more testing. Try with these environments:
- No fuse.py, no libfuse --> Produce error message
- fuse.py, no libfuse --> Produce error message
- no fuse.py, libfuse --> Produce error message
- fuse.py, libfuse --> Should work.
- Try the corresponding commands on a Mac machine.

@kashizui That's a good question, will answer after testing, I'm hoping that fusepy raises an error.

**Comment by maxwang7** [comment]
Also fixes wedit problems -- tested in a Dockerized ubuntu with new CodaLab installation.

**Comment by fabeschan** [comment]
lgtm

**Comment by maxwang7** [comment]
Great! The same CircleCI test that's failing here is also failing on the setup-features branch. I'll wait to merge  this after that one's ready.

**Comment by maxwang7** [comment]
Fixes #760

**Pull request merged**